src/test/java/seedu/address/model/person/NoteTest.java

```diff
@@ -0,0 +1,62 @@
+package seedu.address.model.person;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static seedu.address.testutil.Assert.assertThrows;
+
+import org.junit.jupiter.api.Test;
+
+public class NoteTest {
+
+    @Test
+    public void constructor_null_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> new Note(null));
+    }
+
+    @Test
+    public void equals() {
+        Note note = new Note("Valid Note");
+
+        // same values -> returns true
+        assertTrue(note.equals(new Note("Valid Note")));
+
+        // same object -> returns true
+        assertTrue(note.equals(note));
+
+        // null -> returns false
+        assertFalse(note.equals(null));
+
+        // different types -> returns false
+        assertFalse(note.equals(5.0f));
+
+        // different values -> returns false
+        assertFalse(note.equals(new Note("Other Valid Note")));
+    }
+
+
+    @Test
+    public void isMatch() {
+        Note note = new Note("Likes to eat");
+
+        // Exact match -> returns true
+        assertTrue(note.isMatch("Likes to eat"));
+
+        // Substring whole word -> returns true
+        assertTrue(note.isMatch(" Likes"));
```

Comment:
Should we remove the whitespace in front of `" Likes"`? Each test case should test only 1 thing, whitespaces is being tested in another test case below.

Reply:
Same with the next 2 test cases

Reply:
Great catch 🤠 Fixed